Main.elm

```diff
@@ -14,52 +15,65 @@ import Json.Encode
 type Msg
     = NewEmail String
     | NewPassphrase String
-    | NewData String
+    | NewData (Maybe String)
+    | UpdateContent String
     | NewError String
     | GetData
     | Lock
-    | SetData String
     | DataSaved String
     | DataNotSaved String
-    | UnStored Time.Time
+    | Modifying Time.Time
     | BlurSelection
     | CopySelection
     | ToggleReveal
+    | LastModified Time.Time
 
 
 type alias Model =
     { lock : Bool
     , email : String
     , passphrase : String
     , content : String
-    , stored : Bool
+    , loadedContent :
+        -- may be desynchronized with "content", only used to redraw the
+        -- contentEditable with new decrypted content
+        String
+    , modified : Bool
     , error : String
     , reveal : Bool
+    , lastModified : Maybe Time.Time
     }
 
 
 init : ( Model, Cmd msg )
 init =
-    Model True "" "" "" False "" False ! []
-
-
-
--- Handle contenteditable events
-
-
-innerHtmlDecoder =
-    Json.Decode.at [ "target", "innerHtml" ] Json.Decode.string
+    Model True "" "" "" "" False "" False Nothing ! []
 
 
 
 -- Update
 
 
-update : Msg -> Model -> ( Model, Cmd msg )
+update : Msg -> Model -> ( Model, Cmd Msg )
 update message model =
     case message of
-        UnStored time ->
-            { model | stored = False } ! []
+        Modifying now ->
+            -- The content is being edited, should we save? (aka debounce)
```

Comment:
I think the name `Modifying` and this comment are a bit misleading. IIUC, this event happens every 200 msec after the pad is modified, until it's saved. (A true debounce would cause a save to happen exactly e.g. 1000 msec after the last event, but this one could happen between 1000-1200 msec after the last one.)

I guess there are existing Elm debouncers but we didn't want to spend the effort to integrate with them?